generative-ai/zero-shot-classification.ipynb

```diff
@@ -0,0 +1,271 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Perform zero-shot classification with a foundation model  \n",
+    "  \n",
+    "In this tutorial, we will use an IBM granite model to perform zero-shot classification.  \n",
+    "  \n",
+    "## What is zero-shot classification?\n",
+    "\n",
+    "Zero-shot classification uses zero-shot prompting, a prompt engineering technique that allows a model to perform a task without any specific training or examples. This is an application of zero-shot learning ([ZSL](https://www.ibm.com/topics/zero-shot-learning)) which is a machine learning method that relies on the pre-trained models’ ability to recognize and categorize objects or concepts on post-training. ZSL is similar to few-shot learning ([FSL](https://www.ibm.com/topics/few-shot-learning)), the ability for a model to make an accurate prediction by training on a small number of labeled examples. Both techniques are used to enable models to perform tasks that they haven’t been explicitly trained on.\n",
+    "\n",
+    "Researchers have been experimenting with machine learning models and deep learning algorithms for classification tasks since the 50’s. [The Perceptron](https://courses.cs.washington.edu/courses/cse490h1/19wi/exhibit/machine-learning-1.html#:~:text=Finally%2C%20the%20first%20neural%20network,(History%20of%20the%20Perceptron)) is an early classification model that uses a decision-boundary to classify data into different groups. Most ML/DL [methods](https://www.researchgate.net/publication/359103439_A_Zero-Shot_Learning_Approach_to_Classifying_Requirements_A_Preliminary_Study) rely on supervised learning techniques, and therefore need to be trained using a large amount of task-specific labeled training data. This presents a challenge as the large, annotated datasets required to train these models simply do not exist for every domain. [Some researchers](https://arxiv.org/abs/2312.01044#:~:text=%5B2312.01044%5D%20Large%20Language%20Models%20Are%20Zero%2DShot%20Text%20Classifiers) motivated by these constraints say that LLMs are the way around these data limitations.\n",
```

Comment:
I don't think DL has been defined anywhere?

Reply:
I removed the reference to deep-learning here